pkg/platform/api/api.go

```diff
@@ -0,0 +1,201 @@
+package api
+
+import (
+	"context"
+	"encoding/json"
+	"github.com/bottlerocket-os/bottlerocket-update-operator/pkg/logging"
+	"github.com/pkg/errors"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"time"
+)
+
+const bottlerocketAPISock = "/run/api.sock"
+const minimumRequiredOSVer = "0.3.4"
+
+type updateState = string
+
+const (
+	Idle      updateState = "Idle"
+	Available updateState = "Available"
+	Staged    updateState = "Staged"
+	Ready     updateState = "Ready"
+)
+
+type UpdateImage struct {
+	Arch    string `json:"arch"`
+	Version string `json:"version"`
+	Variant string `json:"variant"`
+}
+
+func (ui *UpdateImage) Identifier() interface{} {
+	return ui.Version
+}
+
+type StagedImage struct {
+	Image      UpdateImage `json:"image"`
+	NextToBoot bool        `json:"next_to_boot"`
+}
+
+type updateCommand = string
+
+const (
+	refresh    updateCommand = "refresh"
+	prepare    updateCommand = "prepare"
+	activate   updateCommand = "activate"
+	deactivate updateCommand = "deactivate"
+)
+
+type commandStatus = string
+
+const (
+	Success commandStatus = "Success"
+	Failed  commandStatus = "Failed"
+	Unknown commandStatus = "Unknown"
+)
+
+type CommandResult struct {
+	CmdType    updateCommand `json:"cmd_type"`
+	CmdStatus  commandStatus `json:"cmd_status"`
+	Timestamp  string        `json:"timestamp"`
+	ExitStatus *int32        `json:"exit_status"`
+	Stderr     *string       `json:"stderr"`
+}
+
+type UpdateStatus struct {
+	UpdateState       updateState    `json:"update_state"`
+	AvailableUpdates  []string       `json:"available_updates"`
+	ChosenUpdate      *UpdateImage   `json:"chosen_update"`
+	ActivePartition   *StagedImage   `json:"active_partition"`
+	StagingPartition  *StagedImage   `json:"staging_partition"`
+	MostRecentCommand *CommandResult `json:"most_recent_command"`
+}
+
+type APIClient struct {
+	log logging.Logger
+	httpClient *http.Client
+}
+
+func NewAPIClient() *APIClient {
```

Comment:
Same question as above for the constructor as well.

```suggestion
func newAPIClient() *apiClient {
```